rholang/src/test/scala/coop/rchain/rholang/interpreter/LetSpec.scala

```diff
@@ -0,0 +1,126 @@
+package coop.rchain.rholang.interpreter
+
+import coop.rchain.rholang.interpreter.errors.{SyntaxError, UnexpectedProcContext}
+import org.scalatest.EitherValues._
+import org.scalatest.{Assertion, Matchers, WordSpec}
+
+/**
+  * This is the first Rholang test class introduced to test a new language construct that
+  * does not test the construct on the normalizer level. That is (1) because it is far more
+  * efficient to write and test Rholang source than to construct normalized Rholang terms
+  * corresponding to the same source, and (2) because this is the first construct that is
+  * implemented in terms of a pre-existing Rholang constructs.
+  */
+class LetSpec extends WordSpec with Matchers {
+
+  "The lexer" should {
+
+    "disallow a non-positive number of let declarations" in {
+      ParBuilderUtil.mkTerm("let in { Nil }").left.value shouldBe a[SyntaxError]
+    }
+
+    "disallow '&' and ';' from appearing in the same let declaration sequence" in {
```

Comment:
just curious - why this is not possible?

Reply:
The Rholang 1.1 spec doesn't include a translation. It's possible once there's a translation. While there's not, we want to make sure the parser catches invalid let declarations.